### ISSUE TITLE
devfile: for a command, allow multiple actions

### ISSUE THREAD

**Issue by themr0c** [eclipse-che/che#17162]
### Describe the bug

Error occurred during creation a workspace from devfile located at `https://raw.githubusercontent.com/themr0c/che-docs/125457461be233851b7b484ffe62e7df5c0b30b9/devfile.yaml`. Cause: Devfile schema validation failed. Error: (/commands/0/actions):The array must have at most 1 element(s), but actual number is 2.

### Che version

https://che.openshift.io

### Steps to reproduce

https://che.openshift.io/f?url=https://github.com/themr0c/che-docs/tree/125457461be233851b7b484ffe62e7df5c0b30b9

The Devfile is containing following:

```
commands:
  - name: Preview server
    actions:
      - type: exec
        component: Tools
        workdir: /projects/che-docs
        command: bash tools/environment_docs_gen.sh
      - type: exec
        command: echo "--modules-folder /tmp/node_modules" > /projects/.yarnrc; export HOME=/projects && export NODE_PATH=/tmp/node_modules && yarn add gulp gulp-cli gulp-connect && LIVERELOAD=true $NODE_PATH/gulp/bin/gulp.js
        component: Antora
        workdir: /projects/che-docs
```

### Expected behavior

In a `devfile`, for a `command`, the list of `actions` should not be limited to one single action.
(Or it would not make sense to have a list).

### Runtime

https://che.openshift.io

**Comment by themr0c** [comment]
Note: we need a sequence of actions running on different containers here, because the antora container does not contain bash. It is something we already do in the Jenkinsfile.

**Comment by sparkoo** [comment]
@themr0c this is not a bug, but a feature. We currently support only 1 action in command. It's documented https://www.eclipse.org/che/docs/che-7/making-a-workspace-portable-using-a-devfile/#che-specific-commands

**Comment by RomanNikitenko** [comment]
I guess some changes are required on task plugin side to support multiple actions, so I added plugins area as well.

**Comment by tsmaeder** [comment]
To clarify: tasks support in che-theia will be broken if we allow multiple actions. I'm all for allowing multiple actions, but before we do, we should have a discussion on how we can map that to tasks in Theia.

**Comment by themr0c** [comment]
I understand it is the documented behavior.

But why is this a list, then? It is confusing.

Would it be possible to convert this issue into a feature request?

As a user, I would like to define a devfile where I can execute one (command) that need different containers to execute:
- a sequence of (actions, steps)?
- parallel (actions, steps)?

Like what is possible with Jenkinsfile stages: it is possible to execute steps in different containers. 

Use cases examples for che-docs:

Sequence: 

- Generate reference tables.
- then Launch preview server.

To have a working implementation, we need 2 commands, and the user must remember to execute one, then the other. It would be better to have a single command executing the 2 actions in sequence:

```
commands:
  - name: Generate reference tables
    actions:
      - type: exec
        component: tools
        workdir: /projects/che-docs
        command: bash tools/environment_docs_gen.sh
  - name: Launch preview server
    actions:
      - type: exec
        command: echo "--modules-folder /tmp/node_modules" > /projects/.yarnrc; export HOME=/projects && export NODE_PATH=/tmp/node_modules && yarn add gulp gulp-cli gulp-connect && LIVERELOAD=true $NODE_PATH/gulp/bin/gulp.js
        component: antora
        workdir: /projects/che-docs
```
Parallel example:

Execute all validations in parallel with one single command:
- Validate that building the documentation runs fine
- Validate the style (execute vale)
- Validate the asciidoc structure (execute test-adoc)

Hybrid example:

- Launch the preview server and keep it running.
- Execute link checker against the preview server.

**Comment by l0rd** [comment]
Please keep in mind that we have been working on a proposal for a new devfile format where `commands` do not have `actions` anymore. Here is a draft of the spec https://devfile.github.io/devfile/_attachments/api-reference.html.

**Comment by ericwill** [comment]
> Please keep in mind that we have been working on a proposal for a new devfile format where `commands` do not have `actions` anymore. Here is a draft of the spec https://devfile.github.io/devfile/_attachments/api-reference.html.

I suppose this ticket is no longer relevant, then?

**Comment by l0rd** [comment]
No it should not be relevant anymore. On v2 of the devfile spec we have composite commands.
